src/utils/IPFSUtils.ts

```diff
@@ -0,0 +1,49 @@
+import PinataClient from "@pinata/sdk";
+import axios from "axios";
+
+/**
+ * Build an IPFS client for interacting with the IPFS API
+ * @param APIKey The project ID
+ * @param secretAPIKey The project secret
+ * @returns An IPFS client for interacting with Pinata
+ */
+export function buildIPFSClient(APIKey: string, secretAPIKey: string): PinataClient {
+  return new PinataClient(APIKey, secretAPIKey);
+}
+
+/**
+ * Retrieves a value from an IPFS gateway
+ * @param contentHash The content hash of the value to retrieve
+ * @param publicGatewayURL The URL of the public IPFS gateway to use
+ * @returns The value, or undefined if it could not be retrieved
+ */
+export async function retrieveValueFromIPFS(
+  contentHash: string,
+  publicGatewayURL: string
+): Promise<string | undefined> {
+  try {
+    const result = await axios.get(`${publicGatewayURL}/ipfs/${contentHash}`, {
+      // We need to set the Accept header to text/plain to avoid
+      // any anomalies with the response
+      headers: {
+        Accept: "text/plain",
+      },
+      // We want just the raw response, not the parsed response
+      transformResponse: (r) => r,
+    });
+    return result.data;
+  } catch (e) {
+    return undefined;
```

Comment:
Hmm, any specific reason not to let an error bubble up? The consumer of the libarary should be the one deciding what to do in case of a fetching error

Reply:
I'm a bit weary of this because the system will error if a key isn't found. However, that's fair that we want to propagate the error so I've removed it. Good callout! https://github.com/across-protocol/sdk-v2/pull/341/commits/421ea066c570643c6900e4272a688b78c73b8de2